src/main/java/org/sagebionetworks/EvaluationUtils.java

```diff
@@ -276,40 +278,79 @@ public static void setStatus(SubmissionStatusModifications statusMods, Submissio
-		if (annotations==null) return;
+		if (annotations!=null) {
+			List<StringAnnotation> sas = annotations.getStringAnnos();
+			if (sas != null) {
+				for (Iterator<StringAnnotation> iterator = sas.iterator(); iterator.hasNext(); ) {
+					StringAnnotation existing = iterator.next();
+					if (existing.getKey().equals(key)) {
+						iterator.remove();
+						return true;
+					}
+				}
+			}
+		}
+		return false;
+	}
 
-		List<StringAnnotation> sas = annotations.getStringAnnos();
-		if (sas!=null) {
-			for (Iterator<StringAnnotation> iterator = sas.iterator(); iterator.hasNext();) {
-				StringAnnotation existing = iterator.next();
-				if (existing.getKey().equals(key)) {
-					iterator.remove();
+	static boolean removeLongAnnotation(SubmissionStatus status, String key) {
+		Annotations annotations = status.getAnnotations();
+		if (annotations!=null) {
+			List<LongAnnotation> las = annotations.getLongAnnos();
+			if (las != null) {
+				for (Iterator<LongAnnotation> iterator = las.iterator(); iterator.hasNext(); ) {
+					LongAnnotation existing = iterator.next();
+					if (existing.getKey().equals(key)) {
+						iterator.remove();
+						return true;
+					}
 				}
 			}
 		}
+		return false;
+	}
 
-		List<DoubleAnnotation> das = annotations.getDoubleAnnos();
-		if (das!=null) {
-			for (Iterator<DoubleAnnotation> iterator = das.iterator(); iterator.hasNext();) {
-				DoubleAnnotation existing = iterator.next();
-				if (existing.getKey().equals(key)) {
-					iterator.remove();
+	static boolean removeDoubleAnnotation(SubmissionStatus status, String key) {
+		Annotations annotations = status.getAnnotations();
+		if (annotations!=null) {
+			List<DoubleAnnotation> das = annotations.getDoubleAnnos();
+			if (das != null) {
+				for (Iterator<DoubleAnnotation> iterator = das.iterator(); iterator.hasNext(); ) {
+					DoubleAnnotation existing = iterator.next();
+					if (existing.getKey().equals(key)) {
+						iterator.remove();
+						return true;
+					}
 				}
 			}
 		}
+		return false;
+	}
 
-		List<LongAnnotation> las = annotations.getLongAnnos();
-		if (las!=null) {
-			for (Iterator<LongAnnotation> iterator = las.iterator(); iterator.hasNext();) {
-				LongAnnotation existing = iterator.next();
-				if (existing.getKey().equals(key)) {
-					iterator.remove();
+	static boolean removeV2Annotation(SubmissionStatus status, String key) {
+		org.sagebionetworks.repo.model.annotation.v2.Annotations annotationsV2 = status.getSubmissionAnnotations();
+		if(annotationsV2 != null) {
+			Map<String, AnnotationsValue> annotationsValueMap = annotationsV2.getAnnotations();
+			if (annotationsValueMap != null) {
+				if (annotationsValueMap.containsKey(key)) {
+					annotationsValueMap.remove(key);
+					annotationsV2.setAnnotations(annotationsValueMap);
+					status.setSubmissionAnnotations(annotationsV2);
+					return true;
 				}
 			}
 		}
+		return false;
+	}
+	
+	static void removeAnnotation(SubmissionStatus status, String key) {
```

Comment:
Could take "Annotations annotations" so that it is clear that it works on old annotations only.